Yoast/wordpress-seo
# Introduce PHP coding-standards thresholds

## Context


* We are working towards using the latest coding standards configuration for a while now. As we are still using PHPCS 0.4.3 we are introducing problems for 2.0.0+ every day because we are unaware that we do so. The checks are simply not being run.
* This PR introduces a way to cap the errors and warnings thresholds and allows us to use the latest configuration to test against.

**Problem to solve:**
We only have branch-checks on Travis, which would mean that we will break trunk if new warnings are introduced.
Enabling trunk builds will require more resources to be used on Travis, but will make this process more reliable.

## Summary


This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Removed `config-yoastcs` as this is being done automatically by the newer YoastCS version
* Removed `maybe-restore-composer.lock` as this is no longer needed
* Added thresholds for CS errors and warnings to the composer configuration; this seemed the most logical place to manage them
* Added the `thresholds` check to the `composer cs` menu

## Test instructions

This PR can be tested by following these steps:

* Use `composer install` to install the latest packages
* Use `composer check-cs-thresholds` to verify the current code is at or below the thresholds
* See Travis running the `check-branch-cs` for the Pull Request instead of doing a full thresholds check

## UI changes

## Documentation

## Quality assurance
